iis-wf/iis-wf-export-actionmanager/src/test/java/eu/dnetlib/iis/wf/export/actionmanager/module/DocumentToSoftwareUrlActionBuilderModuleFactoryTest.java

```diff
@@ -48,43 +48,48 @@
 
     @Before
     public void initModule() {
-        DocumentToSoftwareUrlActionBuilderModuleFactory factory = new DocumentToSoftwareUrlActionBuilderModuleFactory();
-        module = (DocumentToSoftwareUrlActionBuilderModule) factory.instantiate(null, trustLevelThreshold, null);
+        factory = new DocumentToSoftwareUrlActionBuilderModuleFactory();
     }
 
     // ----------------------- TESTS --------------------------
 
     @Test(expected = NullPointerException.class)
-    public void test_build_null_object() throws Exception {
+    public void test_constructor_null_agent() throws Exception {
         // execute
-        module.build(null, agent, actionSetId);
+        factory.new DocumentToSoftwareUrlActionBuilderModule(trustLevelThreshold, null, actionSetId);
```

Comment:
I agree, this is unusual but we are dealing with non-static inner class (relies on `AbstractActionBuilderFactory#algorithmName` which is non-static) so this construct is required.

We could drop this e.g. after making `ActionBuilderModule` `algorithmName` aware but on the other hand this would make each `ActionBuilderModule` more complex.